[cmd/chloggen] `make chlog-new` doesn't work if branch name has '/'

**Describe the bug**

The `make chlog-new` task does not work if running on a branch that has a `/` on it.

**Steps to reproduce**

Create a new branch with a `/` on it:

```
git switch -c example/with-slash-on-it
```

Run `make chlog-new`

**What did you expect to see?**

A new file is created under the `unreleased` folder

**What did you see instead?**

```
❯ make chlog-new
cd cmd/chloggen && go install .
chloggen new -filename example/with-slash-on-it
FAIL: new: open /Users/pablo.baeyens/Source/otel/opentelemetry-collector-contrib/unreleased/example/with-slash-on-it.yaml: no such file or directory
make: *** [chlog-new] Error 1
```

**What version did you use?**

Reproducible as of 9a4ab07e8a6f4a7f755ea57b02bd5b5d58490db9

**What config did you use?**

n/a

**Environment**

I am running on macOS, I would expect this to be a problem on any OS (maybe not Windows)

**Additional context**

Should we escape `/` on the name?

--- a/cmd/chloggen/main.go
+++ b/cmd/chloggen/main.go
@@ -79,7 +79,7 @@ func main() {
 }
 
 func initialize(ctx chlogContext, filename string) error {
-	path := filepath.Join(ctx.unreleasedDir, filename)
+	path := filepath.Join(ctx.unreleasedDir, cleanFileName(filename))
 	var pathWithExt string
 	switch ext := filepath.Ext(path); ext {
 	case ".yaml":
@@ -219,6 +219,11 @@ func deleteEntries(ctx chlogContext) error {
 		}
 	}
 	return nil
+}
+
+func cleanFileName(filename string) string {
+	replace := strings.NewReplacer("/", "_", "\\", "_")
+	return replace.Replace(filename)
 }
 
 func usage() {

--- a/cmd/chloggen/main_test.go
+++ b/cmd/chloggen/main_test.go
@@ -43,6 +43,10 @@ func TestNew(t *testing.T) {
 			filename: "some-change.yml",
 		},
 		{
+			name:     "replace_forward_slash",
+			filename: "replace/forward/slash",
+		},
+		{
 			name:     "bad_extension",
 			filename: "my-change.txt",
 			wantErr:  "non-yaml extension",
@@ -309,4 +313,9 @@ func writeEntryYAML(ctx chlogContext, filename string, entry *Entry) error {
 	}
 	path := filepath.Join(ctx.unreleasedDir, filename)
 	return os.WriteFile(path, entryBytes, os.FileMode(0755))
+}
+
+func TestCleanFilename(t *testing.T) {
+	require.Equal(t, "fix_some_bug", cleanFileName("fix/some_bug"))
+	require.Equal(t, "fix_some_bug", cleanFileName("fix\\some_bug"))
 }

--- a/unreleased/fix_chloggen-slash-branch.yaml
+++ b/unreleased/fix_chloggen-slash-branch.yaml
@@ -1,0 +1,11 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: cmd/chloggen
+
+# A brief description of the change
+note: Fix problem where 'new' command would fail when branch name contained '/'
+
+# One or more tracking issues related to the change
+issues: [11887]

